### ISSUE TITLE
Invalid key length error when using private key

### ISSUE THREAD

**Issue by rgembalik** [git-for-windows/git#1846]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
   64 bit
```
$ git --version --build-options
```

```
git version 2.19.0.windows.1
cpu: x86_64
built from commit: d96bb8bc6c636a8869140e860e72e7bdf64bd790
sizeof-long: 4
sizeof-size_t: 8
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?
Windows 10 Pro 64 bit
```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.17134.285]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: Notepad++
Path Option: Cmd
SSH Option: OpenSSH
CURL Option: OpenSSL
CRLF Option: LFOnly
Bash Terminal Option: ConHost
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled
Enable Builtin Rebase: Disabled
Enable Builtin Stash: Disabled

```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

Hyper-v is enabled, Originally git was installed with symlinks support, but during testing I reinstalled it with symlinks support turned off.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Powershell

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git clone GITLAB_OR_GITHUB_REPO_SSH_ADDRESS
git push -u origin master
```
 - What did you expect to occur after running these commands?

I expect git will connect to origin and download/upload repo commits

 - What actually happened instead?

After providing key password I keeps seeing the following error:
```
Load key "C:\\Users\\USERNAME\\.ssh\\id_rsa": Invalid key length
git@gitlab.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```
The key which causes problems cannot be used from my specific PC. I have the same key on the other PC (which is laptop, and might have older git version, but other than that it's pretty much the same environment) and there it works without any problems. The key can be also opened with puttygen and reexported, but without luck.

The same key with the same command and repository works under intellij idea IDE.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Any repository which uses that specific key for auth.

**Comment by shiftkey** [comment]
@rgembalik are you able to test the SSH connection with `ssh -Tvv [host]` to diagnose why the server might be rejecting your key? `Invalid key length` doesn't really indicate the root cause based on a rudimentary Google search...

**Comment by rgembalik** [comment]
@shiftkey 
Unfortunately, it does not provide any additional info. After I'm asked about key passphrase everything goes as above:
```
Enter passphrase for key 'C:\Users\USERNAME\.ssh\id_rsa':
Load key "C:\\USERNAME\\riven\\.ssh\\id_rsa": Invalid key length
debug2: we did not send a packet, disable method
debug1: No more authentication methods to try.
git@gitlab.com: Permission denied (publickey).
```
However, if I import this key to putty and connect to GitLab, then I see the correct welcome message once passphrase is provided.

**Comment by dscho** [comment]
> The key can be also opened with puttygen and reexported, but without luck.

This suggests that the SSH option "PuTTY" should have been used instead of "OpenSSH".

**Comment by dscho** [comment]
> Load key "C:\\Users\\USERNAME\\.ssh\\id_rsa": Invalid key length

This could mean that the key length is no longer considered safe. Try with `ssh -vvvvvv` to find out more, maybe?

**Comment by rgembalik** [comment]
@dscho Sorry, for no response for such a long time. 
No luck with longer debug for me. A wrong password will cause a new prompt for a password, while correct password ends in what's below. I updated git for windows several times since problem occurred. With no luck though.

Here is longer output for the key:

```
debug2: key: C:\\Users\\USERNAME\\.ssh\\id_rsa (0000000000000000), explicit
debug3: send packet: type 5
debug3: receive packet: type 6
debug2: service_accept: ssh-userauth
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug3: send packet: type 50
debug3: receive packet: type 51
debug1: Authentications that can continue: publickey
debug3: start over, passed a different list publickey
debug3: preferred publickey,keyboard-interactive,password
debug3: authmethod_lookup publickey
debug3: remaining preferred: keyboard-interactive,password
debug3: authmethod_is_enabled publickey
debug1: Next authentication method: publickey
debug1: Trying private key: C:\\Users\\USERNAME\\.ssh\\id_rsa
debug3: failed to open file:D:/dev/tty error:3
debug1: read_passphrase: can't open /dev/tty: No such file or directory
Enter passphrase for key 'C:\Users\riven\.ssh\id_rsa':
Load key "C:\\Users\\USERNAME\\.ssh\\id_rsa": Invalid key length
debug2: we did not send a packet, disable method
debug1: No more authentication methods to try.
git@github.com: Permission denied (publickey).
```

**Comment by landstander668** [comment]
@rgembalik What's the key length? You should be able to check this using the `ssh-keygen -l` command.

If it's less than 1024 bits, OpenSSH will reject the key with an "Invalid key length" error starting at [version 7.6](https://www.openssh.com/txt/release-7.6). I'm not aware of any way to override this behaviour, short of recompiling OpenSSH with a custom SSH_RSA_MINIMUM_MODULUS_SIZE setting... your best bet in this case would be to simply create a new key.

Git for Windows release 2.19.0 shipped with OpenSSH 7.7p1, so this seems like a plausible cause (at least superficially).

**Comment by rgembalik** [comment]
@landstander668 This might be it. Unfortunately, I'm off by one bit. I guess it's one of my oldest keys. I guess it's time to migrate to a longer key already. 

Thank you for your help.
